src/join/core.jl

```diff
@@ -465,19 +465,27 @@ function find_inner_rows(joiner::DataFrameJoiner)
         end
     end
 
-    # this is a workaround for https://github.com/JuliaData/CategoricalArrays.jl/issues/319
-    # this path will be triggered only in rare cases when the refpool code above
-    # fails to convert CategoricalArray into refpool
+    # although isless should be transitive it is safer not to rely on it if we do not have to
```

Comment:
This comment is ambiguous: it could mean that we don't rely on transitivity at all. It's not that we "have to", is that we decide "not to" for nonstandard cases. But we're still not safe if a custom numeric or string types uses an incorrect definition of `isless`.

Reply:
OK - I remove it. I mean "we rely on transitivity and we know of cases when it is not transitive (although it should be) so for safety we disable fast path"